src/org/openlcb/cdi/swing/CdiPanel.java

```diff
@@ -1450,6 +1592,8 @@ protected void writeDisplayTextToNode() {
             byte[] contents = org.openlcb.Utilities.bytesFromHexString((String) textField
                     .getText());
             entry.setValue(new EventID(contents));
+            _changeMade = true;
+            checkForSave();
```

Comment:
I'm worried that this checkForSave will be called many many times which would make restoring an entire node slow.
I have two suggestions:
1) we should ensure that checkForSave only gets called max of once per second or so. Collapse all queued up calls to it.
2) alternatively, we could call checkForSave from whoever invokes writeDisplayTextToNode. There are only two cases, one doing bulk operations (for all of which there should be only one checkForSave after all are done) and the other handling an individual click by the user (for which there should be one checkForSave).

Reply:
writeDisplayTextToNode is called when a button is pressed by a user, nowhere else that I can see.  Removed checkForSave() from updateColor(), probably redundant. I can't see where this might be called rapidly.